sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/dao/room/EntityTypeConverters.kt

```diff
@@ -177,6 +201,77 @@ class EntityTypeConverters {
     }
 }
 
+/**
+ * A client writable copy is considered a new object with only the properties set on the object
+ * that are writable on bridge.  All other fields sent to bridge will be ignored anyways.
+ * @return a ScheduledActivity that can be sent to bridge
+ */
+fun ScheduledActivityEntity.clientWritableCopy(): ScheduledActivity {
+    val schedule = ScheduledActivity()
+    schedule.guid = guid
+    schedule.startedOn = startedOn?.let { org.joda.time.DateTime(it.toEpochMilli()) }
+    schedule.finishedOn = finishedOn?.let { org.joda.time.DateTime(it.toEpochMilli()) }
+    schedule.clientData = clientData?.data
+    return schedule
+}
+
+/**
+ * A bridge metadata copy is considered a new object with only the properties set on the object
+ * that the ArchiveUtil.createMetaDataFile() function is concerned with.
+ * @return a ScheduledActivity that can be pass to the BridgeDataProvider and have the metadata JSON included.
+ */
+fun ScheduledActivityEntity.bridgeMetadataCopy(): ScheduledActivity {
+    var schedule = ScheduledActivity()
+    val immutableFieldMap = HashMap<String, String>()
+
+    scheduledOn?.let {
+        // TODO: mdephillips 10/14/18 better way to convert from 3tenbp LocalDateTime to jodatime DateTime?
+        val dateTime = DateTime.now()
+                .withYear(it.year)
+                .withDayOfYear(it.dayOfYear)
+                .withHourOfDay(it.hour)
+                .withMinuteOfHour(it.minute)
+                .withSecondOfMinute(it.second)
+        val dateTimeStr = ISODateTimeFormat.dateTime().withOffsetParsed().print(dateTime)
+        immutableFieldMap.put("scheduledOn", dateTimeStr)
+    }
+
+    schedulePlanGuid?.let {
+        immutableFieldMap.put("schedulePlanGuid", it)
+    }
+
+    if (immutableFieldMap.isNotEmpty()) {
+        // TODO: mdephillips 10/14/18 sync with josh or dwayne to make scheduledOn and schedulePlanGuid mutable
+        // TODO: mdephillips 10/14/18 so I don't have to do this json non-sense
+        var scheduleJson = "{"
+        immutableFieldMap.keys.forEachIndexed { i, s ->
```

Comment:
For for the update activity API, the only fields needed/read by the server are guid, startedOn and finishedOn. So when converting ScheduledActivityEntity to ScheduledActivity for a REST call, we don't need to do this. If we're passing other fields to another API/method maybe that API/method should accept a ScheduledActivityEntity or something